### PR TITLE
[expo]: Add missing interfaces/classes for AdMob module.

### PR DESCRIPTION
**Pull request by dawnmist**
This module has existed since SDK v21.0.0, but had not been added to the
typescript definitions. The interface changed in SDK v26.0.0, so a split
between v25 and v26 was required.

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes:
https://docs.expo.io/versions/v23.0.0/
https://docs.expo.io/versions/v24.0.0/
https://docs.expo.io/versions/v25.0.0/
https://docs.expo.io/versions/v26.0.0/
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by weswigham** [comment]
@KonstantinKai
@martynaskadisa
@janaagaard75
@ssanchezmarc
@fhelwanger
@umidbekkarimov

Would any of y'all like to review the update?

@RyanCavanaugh It looks like the bot choose to ignore the authors from the original module, and is treating the new version like a new definition and not requesting the original module's reviewers - is this a known bug?

File: types/expo/index.d.ts [hunk 1]
```diff
@@ -269,14 +340,14 @@ export namespace Audio {
         canRecord: false,
         isDoneRecording: false
     } | {
-        canRecord: true,
```
Comment (janaagaard75) [inline types/expo/index.d.ts]: Please remove these indentation changes.

Comment (dawnmist) [reply types/expo/index.d.ts]: Will do - they were unintended, I thought I'd turned off all autoformatting but obviously not.

**Comment by dawnmist** [comment]
I'm not sure what to do about that Travis failure. A newly merged change to React-Native is causing definitions for Expo to fail 'incorrectly extends interface' - but the Expo definitions are what was there for the SDK v25 and v26 releases - they're supposed to map to React Native 0.52 for Expo 25 and React Native 0.54 for Expo 26. The conflict that Travis is reporting is with the React Native 0.55 definitions, which Expo has not yet been updated to match.

Is there a way to specify which version of react-native definitions should be being used?

**Comment by dawnmist** [comment]
Rebased the merge request on master to pick up [the fix from this merge request](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25149) which removes the conflict between ReactNative and Expo's definitions that was recently introduced.

**Review by avocadowastaken — APPROVED** [review]
Seems legit. Thanks for PR 👍

**Comment by rbuckton** [comment]
I think this change may have resulted in failing tests. I'm seeing the following now when trying to build #25356:

```
Testing...
expo failing:
Error: C:/dev/DefinitelyTyped/types/expo/expo-tests.tsx:61:9
ERROR: 61:9   expect  TypeScript@next compile error:
Property 'style' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<AdMobBanner> & Readonly<{ children?: ReactNode; }>...'.
ERROR: 72:9   expect  TypeScript@next compile error:
Property 'style' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<PublisherBanner> & Readonly<{ children?: ReactNode...'.

C:/dev/DefinitelyTyped/types/expo/index.d.ts:74:48
ERROR: 74:48  expect  TypeScript@next compile error:
Cannot find name 'ViewProperties'.
```

**Comment by weswigham** [comment]
> ERROR: 74:48  expect  TypeScript@next compile error:
Cannot find name 'ViewProperties'.

I think you're looking for https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25307 ; though they kept around the old name (and included an update to rename it to ViewProps in expo, for consistency), so it's a lil' odd; maybe a merge went bad?

**Comment by thw0rted** [comment]
I don't know a lot about the PR process here, but how could #25307 be causing `expo` tests to fail?  And why did it not trigger a Travis build?  (These failures are blocking #25327 .)

**Comment by tkrotoff** [comment]
On master, for:
- types/expo/index.d.ts
- types/expo/v23/index.d.ts
- types/expo/v24/index.d.ts

Change this:
```diff
-export interface AdMobBannerProperties extends ViewProperties {
+export interface AdMobBannerProperties extends ViewProps {
```

Can add it to this PR if needed: #25362

**Comment by weswigham** [comment]
Travis runs a PR build on push, and you can't merge unless the build passes, but this repo is so active that from time to time, a number of related or interfering changes can get merged between the green check getting issued and the merge happening, causing a break to appear. I think we'd need a mergebot (and build queue) that tests and lands PRs on command (rather than using the GitHub UI) to avoid it.

**Comment by dawnmist** [comment]
Looks like pull https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25307 for react-native has been merged, which renamed ViewProperties to ViewProps (which is where the style attribute comes from).

I had thought that there were supposed to have been aliases to keep library use compatibility for that in the react-native merge?

**Comment by dawnmist** [comment]
I'll push a merge req to update expo for the #25307 changes (since I already have it checked out).

**Comment by dawnmist** [comment]
@thw0rted Several of the types in Expo extend the ViewProps type in React-Native. The issue was that these two merges happened in parallel to each other. The AdMob types also use the ViewProps from React-Native, and had a difference in the types for SDK v25 and SDK v26 which meant that the v25 types had to be split (in the same way that the v23/v24 ones already were).

At the time when this merge (to add the AdMob interfaces) was written and submitted, ViewProps was called ViewProperties. So the v25 types were all copied from the current index types using ViewProperties, and the new AdMob interfaces were all using ViewProperties.

The React-Native merge appears to have included the renaming of ViewProperties to ViewProps for the Expo interfaces _that were already in master_ - but this meant that the AdMob interfaces (which hadn't been merged yet) and the v25 interfaces (which didn't exist in master at all yet) were not included in the renaming. Both merge requests on their own were valid.

The Travis builds only get run when someone pushes a change to the repo - but the React-Native merge hadn't needed any further changes between the time when this merge was completed and when it then got merged. So the fact that this merge request was introducing additional items that should have also been renamed with the React-Native merge wasn't picked up by the build system.

**Comment by thw0rted** [comment]
> Both merge requests on their own were valid.

Right, but if the first merge conflicts with the second merge, shouldn't the second fail and require manual resolution?  And then, shouldn't the push to resolve the conflicts trigger a new CI run (which then catches the logical error in the resolution)?  Is this maybe a GH problem where two PRs, both green on their own, execute the actual under-the-hood merge in sequence without the appropriate conflict checks, somehow?

**Comment by dawnmist** [comment]
There were several Travis passes. Every commit with a green "tick" where the Bot removed the "fail" tag was a passing build.

The issue with the failures was the same thing. This merge was built against master, but other pull requests were merged *to* master which then made this one require updates.

Travis is only triggered:
* on the current pull request against master
* only when the current pull request is updated.

If master changes _behind_ the current pull request, no additional build is automatically done. If the current pull request is not changed again, you get the situation that happened with 23503 - the pull build had passed against master _at the time of the last change to the pull request_ and would have failed against current master but _that was not detected because no further changes to the pull request were made_.

**Comment by dawnmist** [comment]
@thw0rted It did - and it _passed_ - [here is the Travis Build](https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/371457209?utm_source=github_status&utm_medium=notification). No further changes were made between that build and the merge.

**Pull request merged**